andela/baby-face-cfh
# #153742461 Signin recieve jwt

#### What does this PR do?
It modifies the current authentication (User login) method to use JSON web token. User gets a JWT on successful login.
It also modifies the client side user authentication (sign out inclusive) to use the new implementation as described above.

#### Description of Task to be completed?
Users once authenticated via the login api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication.

#### How should this be manually tested?
- Clone the repository and change directory into it
- Run `npm install` to install project dependencies
- Ensure mongodb is installed and running
- Run `npm start` to launch the app
- Through POSTMAN or any other API testing client, hit the login route (POST)`/api/auth/login` with a registered email and password (this assumes you had previously created a user account through the (POST)`/api/auth/signup` route)
- You should receive a token on the response body of the request.
- Running `npm test` also shows the written tests to verify the new feature
###### Client side testing
- Load the app in the browser after steps 1, 2 and 3 from above
- Sign in to the app using the existing form and ensure a successful login
- Check the console on the page (from the browser developer tools) and run `localStorage.getItem('token')` to verify the token is stored client side.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#153742461

#### Screenshots (if appropriate)
<img width="1118" alt="screen shot 2018-01-04 at 12 09 37" src="https://user-images.githubusercontent.com/30455641/34561062-984aba08-f148-11e7-9f11-f9ccde5198ad.png">
<img width="1429" alt="screen shot 2018-01-04 at 12 11 28" src="https://user-images.githubusercontent.com/30455641/34561073-a0ebafbe-f148-11e7-94b5-f5d3d86c7e82.png">
